staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go

```diff
@@ -279,6 +281,9 @@ func FindMatchingVolume(
 		if GetPersistentVolumeClass(volume) != requestedClass {
 			continue
 		}
+		if vacEnabled && ptr.Deref(volume.Spec.VolumeAttributesClassName, "") != ptr.Deref(claim.Spec.VolumeAttributesClassName, "") {
```

Comment:
does this mean we will potentially bind incompatible volumes if the feature gate is disabled?

Reply:
```go
vacEnabled = utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass)
```
If the feature gate is disabled, the vacEnabled should be false. the binding logic is the same as before.

do you mean the feature-gate is `enabled -> disabled -> re-enabled`?